lib/matplotlib/table.pyi

```diff
@@ -8,7 +8,12 @@ from .transforms import Bbox
 from .typing import ColorType
 
 from collections.abc import Sequence
-from typing import Any, Literal
+from typing import Any, Literal, TYPE_CHECKING
+
+if TYPE_CHECKING:
```

Comment:
Having an `if TYPE_CHECKING:` in a pyi is kind of redundant, as pyi is _only_ used for type checking. 

This does introduce a type-check time requirement on `pandas`, which I don't fully love, but is not _that_ bad. The if/else here does not protect against not having Pandas (and not even sure a try/except would work in a `pyi`... it would at least be unusual)

Reply:
I'll reiterate that I don't think we should be special casing pandas at all, and we should not have a pandas dataframe as a type annotation.   Is there anywhere else in the library that we do this?

Reply:
Should I import the DataFrame method directly?